internal/webhook/auth/validating.go

```diff
@@ -0,0 +1,179 @@
+//nolint:lll
+
+// Constants for invalid group values that should be rejected.
+const (
+	SystemAuthenticatedGroup = "system:authenticated"
+	EmptyGroup               = ""
+)
+
+// Validator implements webhook.AdmissionHandler for Auth validation webhooks.
+// It validates that AdminGroups and AllowedGroups don't contain invalid values.
+type Validator struct {
+	Client  client.Reader
+	Decoder admission.Decoder
+	Name    string
+}
+
+// Assert that Validator implements admission.Handler interface.
+var _ admission.Handler = &Validator{}
+
+// InjectDecoder implements admission.DecoderInjector so the manager can inject the decoder automatically.
+//
+// Parameters:
+//   - d: The admission.Decoder to inject.
+//
+// Returns:
+//   - error: Always nil.
+func (v *Validator) InjectDecoder(d admission.Decoder) error {
+	v.Decoder = d
+	return nil
+}
+
+// SetupWithManager registers the validating webhook with the provided controller-runtime manager.
+//
+// Parameters:
+//   - mgr: The controller-runtime manager to register the webhook with.
+//
+// Returns:
+//   - error: Always nil (for future extensibility).
+func (v *Validator) SetupWithManager(mgr ctrl.Manager) error {
+	hookServer := mgr.GetWebhookServer()
+	hookServer.Register("/validate-auth", &webhook.Admission{
+		Handler:        v,
+		LogConstructor: shared.NewLogConstructor(v.Name),
+	})
+	return nil
+}
+
+// Handle processes admission requests for create and update operations on Auth resources.
+// It validates that AdminGroups and AllowedGroups don't contain invalid values.
+//
+// Parameters:
+//   - ctx: Context for the admission request (logger is extracted from here).
+//   - req: The admission.Request containing the operation and object details.
+//
+// Returns:
+//   - admission.Response: The result of the admission check, indicating whether the operation is allowed or denied.
+func (v *Validator) Handle(ctx context.Context, req admission.Request) admission.Response {
+	log := logf.FromContext(ctx)
+	ctx = logf.IntoContext(ctx, log)
+
+	// Check if decoder is properly injected
+	if v.Decoder == nil {
+		log.Error(nil, "Decoder is nil - webhook not properly initialized")
+		return admission.Errored(http.StatusInternalServerError, errors.New("webhook decoder not initialized"))
+	}
+
+	var resp admission.Response
+
+	switch req.Operation {
+	case admissionv1.Create, admissionv1.Update:
+		resp = v.validateAuthGroups(ctx, req)
+	default:
+		resp.Allowed = true
+	}
+
+	if !resp.Allowed {
+		return resp
+	}
+
+	return admission.Allowed(fmt.Sprintf("Operation %s on %s allowed", req.Operation, req.Kind.Kind))
+}
+
+// validateAuthGroups validates that AdminGroups and AllowedGroups don't contain invalid values.
+func (v *Validator) validateAuthGroups(ctx context.Context, req admission.Request) admission.Response {
+	log := logf.FromContext(ctx)
+
+	// Check if decoder is properly injected
+	if v.Decoder == nil {
```

Comment:
since the nil check is done in the main Handle method, this is likely not needed